### PR TITLE
[14.0] mail_debrand: pass original args to super()

### PR DESCRIPTION
**Pull request by ioans73**
This PR superseds #712 ping @ryanc-me

@pedrobaeza's proposed changes have been implemented and the original PR's conflicts have been fixed

### CODE REVIEW

**Review by fuentes73 — APPROVED** [review]

File: mail_debrand/models/mail_mail.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+# Copyright 2019 Tecnativa - Ernesto Tejeda
+# Copyright 2020 Onestein - Andrea Stirpe
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from odoo import api, models
+
+
+class MailMail(models.AbstractModel):
+    _inherit = "mail.mail"
+
+    # in messages from objects is adding using Odoo that we are going to remove
```
Comment (pedrobaeza) [inline mail_debrand/models/mail_mail.py]: This should be a docstring of the method, not a comment in the middle.

File: mail_debrand/models/mail_mail.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    def create(self, values_list):
+        for index, _value in enumerate(values_list):
+            values_list[index]["body_html"] = self.env[
+                "mail.render.mixin"
+            ].remove_href_odoo(
+                values_list[index].get("body_html", ""),
+                remove_parent=0,
+                remove_before=1,
+            )
+
```
Comment (pedrobaeza) [inline mail_debrand/models/mail_mail.py]: Don't add empty lines inside a method.

File: mail_debrand/models/mail_mail.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+class MailMail(models.AbstractModel):
+    _inherit = "mail.mail"
+
+    # in messages from objects is adding using Odoo that we are going to remove
+
+    @api.model_create_multi
+    def create(self, values_list):
+        for index, _value in enumerate(values_list):
+            values_list[index]["body_html"] = self.env[
```
Comment (pedrobaeza) [inline mail_debrand/models/mail_mail.py]: You don't need to enumerate:
```suggestion
        for vals in values_list:
            vals["body_html"] = self.env[
```

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
Can we avoid to process each of the created mail with this filter? Why not intercepting the rendering of the template that injects the value?

**Comment by ioans73** [comment]
> Can we avoid to process each of the created mail with this filter? Why not intercepting the rendering of the template that injects the value?

Your proposal seems good to me, but I believe the correct thing would be to propose those changes in another PR, right? (at least the part of modifying the way of deleting the text, so that it is not done in the create() method) In order to unblock this and fix the detected bug as soon as possible
What do you think @pedrobaeza ?

**Comment by pedrobaeza** [comment]
I don't think so, as we are trying to "fix" this module, and the proper fix is that one. Note that this module has been "broken" just in this version.

**Comment by ioans73** [comment]
Ok, we will try to implement the proposal, although I don't entirely agree

**Comment by ioans73** [comment]
@pedrobaeza In the end we've chosen to inherit the `_send_prepare_body()` function, as it's prepared precisely for these type of changes

**Comment by pedrobaeza** [comment]
Thanks for the efforts. Do we need then to override now `_render_template` method?

**Comment by ioans73** [comment]
According to our testing, yes, as not all mails pass through `_send_prepare_body()`

**Review by pedrobaeza — APPROVED** [review]
Thanks for all

**Pull request merged**